admin/lib/validator/restart-validator.js

```diff
@@ -0,0 +1,18 @@
+const ValidationError = require('../validation-error')
+const restartValidator = {}
+
+/**
+ * Validate restart reason
+ * @param reason
+ * @param specifyReason
+ * @returns {Boolean}
+ */
+restartValidator.validateReason = (reason, specifyReason) => {
+  const validationError = new ValidationError()
+  if (reason === 'Did not complete' && specifyReason.length === 0) {
+    validationError.addError('didNotCompleteInfo', 'Error: Please specify further information when "Did not complete" option is selected')
```

Comment:
We have usually put error messages in their own file.  This is useful, should a content review be required.  Suggest you consider it.